### PR TITLE
Hunter docodas adjustments

### PR DESCRIPTION
**Pull request by ncberman**
#1001 
made a few minor tweaks and updated tests

Also included are the spell datamines from today
https://www.wowhead.com/classic/news/class-and-spell-changes-on-the-season-of-discovery-phase-5-ptr-346464

### CODE REVIEW

File: tools/database/rune_overrides.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ var RuneOverrides = []*proto.UIRune{
 
 	// Hunter
 	// As of 2024-06-13 Cobra Slayer is being missed by the scraper because the rune engraving ability is missing "Engrave Rune" in the name
-	{Id: 458393, Name: "Engrave Chest - Cobra Slayer", Icon: "spell_nature_guardianward", Type: proto.ItemType_ItemTypeChest, ClassAllowlist: []proto.Class{proto.Class_ClassHunter}},
+	{Id: 458393, Name: "Engrave Chest - Cobra Slayer", Icon: "spell_nature_guardianward", Type: proto.ItemType_ItemTypeHands, ClassAllowlist: []proto.Class{proto.Class_ClassHunter}},
```
Comment (kayla-glick) [inline tools/database/rune_overrides.go:43]: The name still says chest 😄

File: sim/hunter/runes.go [hunk 2]
```diff
@@ -382,16 +383,23 @@ func (hunter *Hunter) applyHitAndRun() {
...
+				hunter.Unit.MoveSpeed *= 1 / 1.30
 			},
 		})
 	}
 }
+
+func (hunter *Hunter) applyImprovedVolley() {
+	if hunter.HasRune(proto.HunterRune_RuneCloakImprovedVolley) {
+		// The 3% rAP scaling is applied inside the volley spell config itself
+		hunter.Volley.DamageMultiplier *= 2
```
Comment (kayla-glick) [inline sim/hunter/runes.go:403]: I'm not sure whether or not Volley will actually be initialized at this point. Usually i have to make these changes inside an aura

File: sim/hunter/runes.go [hunk 3]
```diff
@@ -248,10 +249,10 @@ func (hunter *Hunter) applyLockAndLoad() {
...
 	lockAndLoadMetrics := hunter.Metrics.NewResourceMetrics(core.ActionID{SpellID: 415413}, proto.ResourceType_ResourceTypeMana)
 
-	icd := core.Cooldown{
-		Timer:    hunter.NewTimer(),
-		Duration: time.Second * 8,
-	}
+	// icd := core.Cooldown{
+	// 	Timer:    hunter.NewTimer(),
+	// 	Duration: time.Second * 8,
+	// }
```
Comment (kayla-glick) [inline sim/hunter/runes.go:255]: Didn't the patch notes says lock and load now has an 8s ICD?

File: tools/database/rune_overrides.go (hunk 1 above)
Comment (ncberman) [reply tools/database/rune_overrides.go:43]: oops

File: sim/hunter/runes.go (hunk 3 above)
Comment (ncberman) [reply sim/hunter/runes.go:255]: Icd was removed as of the latest ptr build

File: sim/hunter/item_sets_pve.go [hunk 4]
```diff
@@ -200,17 +200,21 @@ var ItemSetDragonstalkerProwess = core.NewItemSet(core.ItemSet{
...
-						spell.DamageMultiplier *= 1.20
+						if spell.SpellCode != (SpellCode_HunterRaptorStrikeHit | SpellCode_HunterRaptorStrike) {
+							spell.DamageMultiplier *= 1.20
+						}
 					}
 				},
 				OnExpire: func(aura *core.Aura, sim *core.Simulation) {
 					for _, spell := range hunter.MeleeSpells {
-						spell.DamageMultiplier /= 1.20
+						if spell.SpellCode != (SpellCode_HunterRaptorStrikeHit | SpellCode_HunterRaptorStrike) {
```
Comment (kayla-glick) [inline sim/hunter/item_sets_pve.go:215]: The base raptor strike spell doesn't actually deal any damage so we should be able to just use `SpellCode_HunterRaptorStrikeHit` here

**Review by kayla-glick — APPROVED** [review]

**Pull request merged**